code/datums/diseases/advance/advance.dm

```diff
@@ -385,7 +400,7 @@
 				if(!D.HasSymptom(S))
 					D.symptoms += S
 					i -= 1
-	while(i > 0)

```

Comment:
@Iamgoofball of course it does? the ```do``` means the block of code always runs once (which the define is non-zero so it should always run atleast ONCE), moving the while to the top would mean the condition (```i > 0```) would be checked ```SYMPTOM_LIMIT``` times, where as with the do the condition is only checked ```SYMPTOM_LIMIT-1``` times.